api4/user_test.go

```diff
@@ -20,23 +20,14 @@ import (
 func TestCreateUser(t *testing.T) {
 	th := Setup().InitBasic()
 	defer th.TearDown()
-	Client := th.Client
-	AdminClient := th.SystemAdminClient
-
-	enableOpenServer := th.App.Config().TeamSettings.EnableOpenServer
-	enableUserCreation := th.App.Config().TeamSettings.EnableUserCreation
-	defer func() {
-		th.App.UpdateConfig(func(cfg *model.Config) { cfg.TeamSettings.EnableOpenServer = enableOpenServer })
-		th.App.UpdateConfig(func(cfg *model.Config) { cfg.TeamSettings.EnableUserCreation = enableUserCreation })
```

Comment:
Why don't we need now to restore config after testing?

Reply:
The `api4` and `app` tests star the app with a cloned config file to avoid propagating changes to the subsequent tests:
https://github.com/mattermost/mattermost-server/blob/ae76d27b7dbb8f20b35f17153fe77beb8c0d2297/api4/apitestlib.go#L67-L83

I have a follow up ticket to avoid ever using the local `config/config.json`, since a developer's current configuration can skew the unit tests unexpectedly as compared to a CI build with the default config.